runtime/include/QuantumDevice.hpp

```diff
@@ -148,17 +148,53 @@ struct QuantumDevice {
      * @param inverse Indicates whether to use inverse of gate
      */
     virtual void NamedOperation(const std::string &name, const std::vector<double> &params,
-                                const std::vector<QubitIdType> &wires, bool inverse) = 0;
+                                const std::vector<QubitIdType> &wires, bool inverse)
+    {
+        NamedOperation(name, params, wires, inverse, {}, {});
+    }
+
+    /**
+     * @brief Apply a single gate to the state vector of a device with its name if this is
+     * supported.
+     *
+     * @param name The name of the gate to apply
+     * @param params Optional parameter list for parametric gates
+     * @param wires Wires to apply gate to
+     * @param inverse Indicates whether to use inverse of gate
+     * @param controlled_wires Controlled wires applied to the operation
+     * @param controlled_values Controlled values applied to the operation
```

Comment:
Aren't these args optional too?
```suggestion
     * @param inverse Indicates whether to use inverse of gate
     * @param controlled_wires Optional controlled wires applied to the operation
     * @param controlled_values Optional controlled values applied to the operation
```
it's minor and feel free to ignore this suggestion if you think otherwise.